features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java

```diff
@@ -165,6 +152,10 @@ private Optional<SQLToken> getColumnPositionToken(final EncryptTable encryptTabl
     private Collection<SQLToken> getChangeColumnTokens(final EncryptTable encryptTable, final Collection<ChangeColumnDefinitionSegment> segments) {
         Collection<SQLToken> result = new LinkedList<>();
         for (ChangeColumnDefinitionSegment each : segments) {
+            String columnName = each.getPreviousColumn().getIdentifier().getValue();
+            if (encryptTable.isEncryptColumn(columnName)) {
+                throw new UnsupportedOperationException("Unsupported operation 'change' for the cipher column");
```

Comment:
Is it better to use ShardingSpherePreconditions?

Reply:
THX for point out.